txpipe/metadata.py

```diff
@@ -0,0 +1,124 @@
+            copy(shear_tomo_file, "response", "tracers", "R")
+            copy(shear_tomo_file, "response", "tracers", "K")
+            copy(shear_tomo_file, "response", "tracers", "C")
+
+        copy(shear_tomo_file, "tomography", "tracers", "N_eff")
+        copy(lens_tomo_file, "tomography", "tracers", "lens_counts")
+        copy(shear_tomo_file, "tomography", "tracers", "sigma_e")
+        copy(shear_tomo_file, "tomography", "tracers", "source_counts")
+
+        N_eff = shear_tomo_file["tomography/N_eff"][:]
+        n_eff = N_eff / area_sq_arcmin
+
+        lens_counts = lens_tomo_file["tomography/lens_counts"][:]
+        source_counts = shear_tomo_file["tomography/source_counts"][:]
+
+        lens_density = lens_counts / area_sq_arcmin
+        source_density = source_counts / area_sq_arcmin
+
+        meta_file.create_dataset("tracers/n_eff", data=n_eff)
+        meta_file.create_dataset("tracers/lens_density", data=lens_density)
+        meta_file.create_dataset("tracers/source_density", data=source_density)
+        meta_file["tracers"].attrs["area"] = area
+        meta_file["tracers"].attrs["area_unit"] = "deg^2"
+        meta_file["tracers"].attrs["density_unit"] = "arcmin^{-2}"
+        copy_attrs(shear_tomo_file, "tomography", "tracers")
+        copy_attrs(lens_tomo_file, "tomography", "tracers")
+
+        meta_file.close()
+
+        # human readable version
+        yaml_out_name = self.get_output("tracer_metadata_yml")
+        metadata = {
+            "lens_density": lens_density.tolist(),
+            "source_density": source_density.tolist(),
+            "sigma_e": shear_tomo_file["tomography/sigma_e"][:].tolist(),
+            "n_eff": n_eff.tolist(),
+            "lens_counts": lens_counts.tolist(),
+            "source_counts": source_counts.tolist(),
+            "area": float(area),
+            "area_unit": "deg^2",
+            "density_unit": "arcmin^{-2}",
+        }
+
+        if shear_catalog_type == "metacal":
+            metadata["R_gamma_mean"] = (
+                shear_tomo_file["metacal_response/R_gamma_mean"][:].tolist(),
+            )
+            metadata["R_S"] = (shear_tomo_file["metacal_response/R_S"][:].tolist(),)
+            metadata["R_total"] = (
+                shear_tomo_file["metacal_response/R_total"][:].tolist(),
+            )
+        else:
+            metadata["R"] = (shear_tomo_file["response/R"][:].tolist(),)
+            metadata["K"] = (shear_tomo_file["response/K"][:].tolist(),)
+            metadata["C"] = (shear_tomo_file["response/C"][:].tolist(),)
+
+        f = open(yaml_out_name, "w")
+        yaml.dump(metadata, f)
+        f.close()
+
+        shear_tomo_file.close()
+        lens_tomo_file.close()
+
+    def read_area(self):
+        with self.open_input("mask", wrapper=True) as f:
+            m = f.read_map("mask")
+        print(m.max())
+        f_sky = (m > 0).sum() / m.size
+        area_sq_deg = f_sky * 41252.96125
+        print(area_sq_deg)
```

Comment:
remove or add something to say what this number is

Reply:
Done.